draft-ietf-jsonpath-base.md

```diff
@@ -1073,7 +1071,7 @@ The `descendant-selector` selects certain descendants of a node:
 In the resultant nodelist:
 
 * nodes occur before their children, and
-* nodes of an array occur in array order.
+* nodes of a given array, and descendants of nodes of the given array, occur in array order.
```

Comment:
Is this specific enough to clarify between the two cases I mentioned in the issue?

- complete each item and its descendants before moving to the next item, in array order, vs
- complete the items in the array, in order, then cycle through descendants of members (queue-style)

Reply:
It might be worth including the `[[0],[1]]` example.

Reply:
> Is this specific enough to clarify between the two cases I mentioned in the issue?
> 
>     * complete each item and its descendants before moving to the next item, in array order, vs
> 
>     * complete the items in the array, in order, then cycle through descendants of members (queue-style)

Yes. The second implementation is unlikely to work without some clever/complex handling of the nodelist. (The reason I say this is that the traversal order doesn't necessarily determine the order of the resultant nodelist. For example, after traversal is complete, the nodelist could be sorted into an order consistent with the spec rules. Not a good implementation, but possible.)